### ISSUE TITLE
[BUG] (Resolved) Error: no matching Route53Zone found

### ISSUE THREAD

**Issue by usrbinkat** [CodeSparta/devkit-vpc#11]
# Issue: terraform fails at no matching Route53Zone found    
Impact: all active Sparta demos & deliveries likely impacted    
Expected: expected terraform to complete vpc creation without errors    
Actual outcome: terraform fails on error & requires manual intervention to create Route53Zone which breaks TF state    

--------------------------------------------------------------------------------------------------
#### Steps to reproduce:
```
git clone https://github.com/CodeSparta/devkit-vpc.git
cd devkit-vpc
vi variables.tf
source tools/dev.sh
./site.yml -vvv -e aws_cloud_region=us-gov-west-1 \
  -e aws_access_key=AKXXXXXXXXXXXXXXNEU \
  -e aws_secret_key=NasXXXXXXXXXXXXXXXXB6vK
```
#### Then to manually exec terraform and reproduce error directly:
```
terraform apply -var-file=global.tfvars
```
## Error:
```
root@64838c5d393b devkit-vpc$ terraform apply -var-file=global.tfvars
module.bastion-node.random_id.index: Refreshing state... [id=V10]
module.vpc.data.aws_vpc_endpoint_service.ec2: Refreshing state... [id=3577956811]
module.vpc.aws_vpc.cluster_vpc: Refreshing state... [id=vpc-00ed5c865d42de2ec]
module.vpc.data.aws_vpc_endpoint_service.elasticloadbalancing: Refreshing state... [id=1355104433]
module.vpc.data.aws_vpc_endpoint_service.s3: Refreshing state... [id=1536277269]
module.iam-roles.aws_iam_role.master_role: Refreshing state... [id=validate-458-master-role]
module.iam-roles.aws_iam_role.worker_role: Refreshing state... [id=validate-458-worker-role]
module.iam-roles.aws_iam_role_policy.master_policy: Refreshing state... [id=validate-458-master-role:validate-458-master-policy]
module.iam-roles.aws_iam_instance_profile.master: Refreshing state... [id=validate-458-master-profile]
module.iam-roles.aws_iam_instance_profile.worker: Refreshing state... [id=validate-458-worker-profile]
module.iam-roles.aws_iam_role_policy.worker_policy: Refreshing state... [id=validate-458-worker-role:validate-458-worker-policy]
module.vpc.aws_route_table.pri_route_table: Refreshing state... [id=rtb-0d1209b4672bdc2cc]
module.bastion-node.data.aws_vpc.cluster_vpc: Refreshing state... [id=vpc-00ed5c865d42de2ec]
module.vpc.aws_subnet.pri_subnet[2]: Refreshing state... [id=subnet-0a49fa5be098a0c78]
module.vpc.aws_security_group.private_ec2_api: Refreshing state... [id=sg-04e0f5f0f002e5533]
module.vpc.aws_vpc_endpoint.private_s3: Refreshing state... [id=vpce-099461d9d124e6633]
module.vpc.aws_security_group.private_elb_api: Refreshing state... [id=sg-0ba59a282004d217b]
module.vpc.aws_subnet.public-subnet[2]: Refreshing state... [id=subnet-0ab9805b4be037631]
module.vpc.aws_subnet.pri_subnet[1]: Refreshing state... [id=subnet-0545e2549e5352b96]
module.vpc.aws_subnet.pri_subnet[0]: Refreshing state... [id=subnet-03cfe31689eba69d2]
module.vpc.aws_internet_gateway.inet-gateway[0]: Refreshing state... [id=igw-0e655fae559401e58]
module.vpc.aws_subnet.public-subnet[0]: Refreshing state... [id=subnet-02f0aee94f5c79eaa]
module.security-groups.data.aws_vpc.cluster_vpc: Refreshing state... [id=vpc-00ed5c865d42de2ec]
module.route-53.data.aws_vpc.cluster_vpc: Refreshing state... [id=vpc-00ed5c865d42de2ec]
module.vpc.aws_subnet.public-subnet[1]: Refreshing state... [id=subnet-0afbd9d4a5a4dea64]
module.vpc.aws_security_group_rule.private_elb_api_egress: Refreshing state... [id=sgrule-1512807283]
module.vpc.aws_security_group_rule.private_ec2_api_egress: Refreshing state... [id=sgrule-3216256728]
module.vpc.aws_security_group_rule.private_ec2_ingress: Refreshing state... [id=sgrule-179686812]
module.vpc.aws_security_group_rule.private_elb_ingress: Refreshing state... [id=sgrule-1263806537]
module.vpc.aws_route_table.public-route-table[2]: Refreshing state... [id=rtb-00a1a584e1258562d]
module.vpc.aws_route_table.public-route-table[0]: Refreshing state... [id=rtb-0466b54d3e43d5ea5]
module.vpc.aws_route_table.public-route-table[1]: Refreshing state... [id=rtb-087bf9339c32b0e0b]
module.vpc.aws_vpc_endpoint.elasticloadbalancing: Refreshing state... [id=vpce-08075db2d2ad1f28c]
module.vpc.data.aws_subnet.public[0]: Refreshing state... [id=subnet-03cfe31689eba69d2]
module.vpc.data.aws_subnet.public[1]: Refreshing state... [id=subnet-0545e2549e5352b96]
module.vpc.data.aws_subnet.public[2]: Refreshing state... [id=subnet-0a49fa5be098a0c78]
module.vpc.data.aws_subnet.private[2]: Refreshing state... [id=subnet-0a49fa5be098a0c78]
module.vpc.data.aws_subnet.private[0]: Refreshing state... [id=subnet-03cfe31689eba69d2]
module.vpc.data.aws_subnet.private[1]: Refreshing state... [id=subnet-0545e2549e5352b96]
module.vpc.aws_vpc_endpoint.private_ec2: Refreshing state... [id=vpce-0d3a5f7fb1c753fdb]
module.vpc.aws_route_table_association.pri_net_route_table_assoc[1]: Refreshing state... [id=rtbassoc-026e57287bca2aeff]
module.vpc.aws_route_table_association.pri_net_route_table_assoc[2]: Refreshing state... [id=rtbassoc-0600a6ec1d23d4dad]
module.vpc.aws_route_table_association.pri_net_route_table_assoc[0]: Refreshing state... [id=rtbassoc-09476c594aaccf0e9]
module.vpc.aws_vpc_endpoint_route_table_association.private_s3[0]: Refreshing state... [id=a-vpce-099461d9d124e66333784233196]
module.vpc.aws_vpc_endpoint_route_table_association.private_s3[1]: Refreshing state... [id=a-vpce-099461d9d124e66333784233196]
module.vpc.aws_vpc_endpoint_route_table_association.private_s3[2]: Refreshing state... [id=a-vpce-099461d9d124e66333784233196]
module.bastion-node.data.aws_subnet_ids.public: Refreshing state... [id=vpc-00ed5c865d42de2ec]
module.vpc.aws_route_table_association.public_route_table_assoc: Refreshing state... [id=rtbassoc-0fa24007fa183622e]
module.vpc.aws_route_table_association.route_net[0]: Refreshing state... [id=rtbassoc-0fa24007fa183622e]
module.vpc.aws_route_table_association.route_net[1]: Refreshing state... [id=rtbassoc-0286fcd7d09cbf74a]
module.vpc.aws_route_table_association.route_net[2]: Refreshing state... [id=rtbassoc-0e704aebabd081c44]
module.security-groups.aws_security_group.registry-sg: Refreshing state... [id=sg-0edbe4c3f93c62fb0]
module.security-groups.aws_security_group.worker-sg: Refreshing state... [id=sg-00b12d7a38af59496]
module.security-groups.aws_security_group.master-sg: Refreshing state... [id=sg-03beced027f804a54]
module.route-53.aws_route53_zone.private_zone: Refreshing state... [id=Z091825816SNN2X1U3YT0]
module.security-groups.aws_security_group_rule.registry_egress_all: Refreshing state... [id=sgrule-341844756]
module.security-groups.aws_security_group_rule.registry_ingress_5000: Refreshing state... [id=sgrule-3576874239]
module.security-groups.aws_security_group_rule.registry_ingress_8080: Refreshing state... [id=sgrule-670679657]
module.security-groups.aws_security_group_rule.registry_ingress_22: Refreshing state... [id=sgrule-948451796]
module.security-groups.aws_security_group_rule.worker_ingress_all: Refreshing state... [id=sgrule-871112064]
module.security-groups.aws_security_group_rule.worker_egress_all: Refreshing state... [id=sgrule-1418914984]
module.security-groups.aws_security_group_rule.master_ingress_all: Refreshing state... [id=sgrule-1734229834]
module.security-groups.aws_security_group_rule.master_egress_all: Refreshing state... [id=sgrule-83728080]
module.registry-node.data.aws_vpc.cluster_vpc: Refreshing state... [id=vpc-00ed5c865d42de2ec]
module.registry-node.random_id.index: Refreshing state... [id=cnk]
module.registry-node.data.aws_subnet_ids.public: Refreshing state... [id=vpc-00ed5c865d42de2ec]
module.registry-node.aws_instance.registry-node: Refreshing state... [id=i-058e575d1ecabc164]

Warning: Interpolation-only expressions are deprecated

  on route-53/private-dns.tf line 2, in resource "aws_route53_zone" "private_zone":
   2:   name =  "${var.cluster_domain}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.

Error: no matching Route53Zone found
root@64838c5d393b devkit-vpc$
```

## Cleanup VPC with:
```
terraform destroy -var-file=global.tfvars
```

**Comment by day0hero** [comment]
@usrbinkat see #12

**Comment by usrbinkat** [comment]
resolved in #12 tested & confirmed fix
